pkg/detectors/launchdarkly/launchdarkly.go

```diff
@@ -47,25 +59,48 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 		}
 
 		if verify {
-			req, err := http.NewRequestWithContext(ctx, "GET", "https://app.launchdarkly.com/api/v2/tokens", nil)
-			if err != nil {
-				continue
-			}
-			req.Header.Add("Authorization", resMatch)
-			res, err := client.Do(req)
-			if err == nil {
-				defer res.Body.Close()
-				if res.StatusCode >= 200 && res.StatusCode < 300 {
-					s1.Verified = true
-				} else {
-					// This function will check false positives for common test words, but also it will make sure the key appears 'random' enough to be a real key.
-					if detectors.IsKnownFalsePositive(resMatch, detectors.DefaultFalsePositives, true) {
-						continue
+			if strings.HasPrefix(resMatch, "api-") {
+				req, err := http.NewRequestWithContext(ctx, "GET", "https://app.launchdarkly.com/api/v2/tokens", nil)
+				if err != nil {
+					continue
+				}
+				client := s.client
+				if client == nil {
+					client = defaultClient
+				}
+				req.Header.Add("Authorization", resMatch)
+				res, err := client.Do(req)
+				if err == nil {
+					defer res.Body.Close()
+					if res.StatusCode >= 200 && res.StatusCode < 300 {
+						s1.Verified = true
+					} else if res.StatusCode == 401 {
+						// 401 is expected for an invalid token, so there is nothing to do here.
+					} else {
+						s1.VerificationError = fmt.Errorf("unexpected HTTP response status %d", res.StatusCode)
 					}
+				} else {
+					s1.VerificationError = err
+				}
+			} else {
+				// This is a server SDK key. Try to initialize using the SDK.
+				_, err := ldclient.MakeCustomClient(resMatch, defaultSDKConfig, defaultSDKTimeout)
+				if err == nil {
+					s1.Verified = true
+				} else if err == ldclient.ErrInitializationFailed {
+					// If initialization fails, the key is not valid, so do nothing.
+				} else if err == ldclient.ErrInitializationTimeout {
+					// If initialization times out, we don't know if the key is valid.
+					s1.VerificationError = err
```

Comment:
How do you want to handle errors that are neither of these named cases?

Reply:
Ah, the documentation claims that those are the only two errors it can return. But reading it immediately contradicts that. So I think all non-ErrInitializationFailed are likely in the "don't know" case. Though it looks like there's a "SDK key contains invalid characters" that definitely means it is not verified.